charmhelpers/contrib/openstack/context.py

```diff
@@ -2798,6 +2798,18 @@ def __init__(self, bridges_key=None, bonds_key=None, enable_dpdk_key=None,
                 self._mac_pci_address_map[bond_mac.mac] = pci_address
 
         config_bridges = config(bridges_key) or ''
+        # check config_bridges pattern "<str>:<str>","<str>:<str> <str>:<str>" or ""
+        config_bridges_pattern = re.match(
+            r'^((\S+:\S+)( \S+:\S+)*|^$)$',
```

Comment:
I'm not too happy about the use of regular expressions in this context. Reading the original bug, Pythons text parsing functions used below would raise an ValueError already, and looking at an charm implementation consuming this change:

https://github.com/openstack-charmers/charm-layer-ovn/pull/50/commits/1d2a5a61c6289c3831e893aed3044e8d20be9c42#diff-727089e91d7a493a8b32c81042f625eed48f5b67ca485f26469a85d078eaa353R785-R793

You are adding exception handling to the charm implementation anyway. What do you think about just omitting the regex?

Reply:
My idea behind the regular expression it was to be more restrictive to what we consider as an bad input, but looking again the `config_bridges.split()` does an amazing job of removing white spaces between the key-values, been more fault tolerant with the user input.

The only "real gain" that I can see now using regex would be in a situation where the user doesn't give the `ifname_or_mac` like `br-internet:` that would return an empty string to `ifname_or_mac`, probably adding an unmapped interface which I think it's not a big deal.

Thanks for your view. I do agree that omitting the regex is a good idea.